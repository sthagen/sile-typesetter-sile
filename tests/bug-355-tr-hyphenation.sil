\begin[papersize=a7]{document}
\use[module=packages.retrograde,target=v0.15.0]
\nofolios
\neverindent
\set[parameter=shaper.variablespaces, value=false]
\set[parameter=linebreak.emergencyStretch,value=0]
\font[size=20pt,language=tr]
% Without stretchable spaces there is no variablility for SILE to do justification.
% As as a result *most* lines in this test are expected to overflow.
% What we hope to see is:
% * The first available breakpoint being used after an overflow.
% * ... even when that breakpoint is a special discretionary apostrophe.

\kern[width=170pt]İsa'ya
\kern[width=147pt]İsa'ya

% Same test but with curly apostrophes, both should work as discretionary breaks.

\kern[width=170pt]İsa’ya
\kern[width=147pt]İsa’ya

% Test that hyphenation continues to happen in long words that already broke a line.
Afyonkarahisarlılaştırabildiklerimizdenmişsinizcesine.

\end{document}
