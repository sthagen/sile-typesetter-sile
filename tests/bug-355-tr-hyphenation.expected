Set paper size 	209.7637818	297.6377985
Begin page
Mx 	180.4882
My 	30.8585
Set font 	Gentium Plus;20;400;;normal;;;LTR
T	997 86 68 w=22.8613	(İsa)
Mx 	203.3495
T	10 w=5.0000	(')
Mx 	10.4882
My 	54.8585
T	92 68 w=19.0430	(ya)
Mx 	180.3482
T	997 86 68 w=22.8613	(İsa)
Mx 	203.2095
T	10 w=5.0000	(')
Mx 	10.4882
My 	78.8585
T	92 68 w=19.0430	(ya)
Mx 	180.4882
My 	102.8585
T	997 86 68 w=22.8613	(İsa)
Mx 	203.3495
T	183 w=4.5801	(’)
Mx 	10.4882
My 	126.8585
T	92 68 w=19.0430	(ya)
Mx 	180.3482
T	997 86 68 w=22.8613	(İsa)
Mx 	203.2095
T	183 w=4.5801	(’)
Mx 	10.4882
My 	150.8585
T	92 68 w=19.0430	(ya)
My 	174.8585
T	36 73 w=18.1250	(Af)
Mx 	28.6132
T	92 82 81 w=30.9570	(yon)
Mx 	59.5702
T	78 68 w=18.9355	(ka)
Mx 	78.5058
T	85 68 w=17.0996	(ra)
Mx 	95.6054
T	75 76 w=16.4551	(hi)
Mx 	112.0605
T	86 68 85 w=24.8242	(sar)
Mx 	136.8847
T	79 215 w=10.8398	(lı)
Mx 	147.7245
T	79 68 1578 w=22.3242	(laş)
Mx 	170.0487
T	87 215 w=12.3047	(tı)
Mx 	182.3534
T	85 68 w=17.0996	(ra)
Mx 	199.4530
T	16 w=6.7383	(-)
Mx 	10.4882
My 	198.8585
T	69 76 79 w=20.9961	(bil)
Mx 	31.4843
T	71 76 78 w=25.5762	(dik)
Mx 	57.0605
T	79 72 w=14.6582	(le)
Mx 	71.7187
T	85 76 w=13.3398	(ri)
Mx 	85.0585
T	80 76 93 w=29.9707	(miz)
Mx 	115.0292
T	71 72 81 w=30.6738	(den)
Mx 	145.7030
T	80 76 1578 w=29.2090	(miş)
Mx 	174.9120
T	86 76 w=13.1445	(si)
Mx 	188.0565
T	81 76 93 w=24.9414	(niz)
Mx 	212.9980
T	16 w=6.7383	(-)
Mx 	10.4882
My 	222.8585
T	70 72 w=17.9785	(ce)
Mx 	28.4667
T	86 76 81 72 w=33.4180	(sine)
Mx 	61.8847
T	17 w=4.5801	(.)
End page
Finish
